### PR TITLE
gc(ticdc):  do not block gc if there is no changefeed

### LINKED ISSUE #9633: ticdc still block gc after all changefeeds are removed
if there is no changefeed, ticdc should remove the gc service from PD

reproduce steps:
1.  add a changefeed 
2. check the gc service checkpoint 
```shell
 curl http://127.0.0.1:2379/pd/api/v1/gc/safepoint
{
  "service_gc_safe_points": [
    {
      "service_id": "gc_worker",
      "expired_at": 9223372036854775807,
      "safe_point": 443738300532916224
    },
    {
      "service_id": "ticdc-default-15674009460217235928",
      "expired_at": 1692814223,
      "safe_point": 443738442313498625
    }
  ],
  "gc_safe_point": 443738127517876224
}
```
4. remove the changefeed
5. check the gc service checkpoint again
```shell
curl http://127.0.0.1:2379/pd/api/v1/gc/safepoint
{
  "service_gc_safe_points": [
    {
      "service_id": "gc_worker",
      "expired_at": 9223372036854775807,
      "safe_point": 443739841939636224
    },
    {
      "service_id": "ticdc-default-15674009460217235928",
      "expired_at": 1692814223,
      "safe_point": 443738442313498625
    }
  ],
  "gc_safe_point": 443738442313498625
}
```

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #9633

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/controller/controller.go [hunk 1]
```diff
@@ -257,6 +258,22 @@ func (o *controllerImpl) calculateGCSafepoint(state *orchestrator.GlobalReactorS
...
 		}
 	}
+
+	// check if the upstream has a changefeed, if not we should update the gc safepoint
+	_ = o.upstreamManager.Visit(func(up *upstream.Upstream) error {
+		hasGcServiceSafePoint := false
+		if _, exist := minCheckpointTsMap[up.ID]; exist {
+			hasGcServiceSafePoint = true
+		}
+		if _, exist := forceUpdateMap[up.ID]; exist {
```
Comment (CharlesCheung96) [inline cdc/controller/controller.go:268]: This is a redundant judgment since all upstreamID related information is stored in minCheckpointTsMap.

**Review by asddongmen — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**